src/spec-lang/tc/builtins.ts

```diff
@@ -1,143 +1,178 @@
         )
     ],
     [
         "addmod",
-        new SFunctionType(
-            [new SIntType(256, false), new SIntType(256, false), new SIntType(256, false)],
-            [new SIntType(256, false)],
+        new FunctionType(
+            undefined,
+            [new IntType(256, false), new IntType(256, false), new IntType(256, false)],
+            [new IntType(256, false)],
             FunctionVisibility.Default,
             FunctionStateMutability.Pure
         )
     ],
     [
         "mulmod",
-        new SFunctionType(
-            [new SIntType(256, false), new SIntType(256, false), new SIntType(256, false)],
-            [new SIntType(256, false)],
+        new FunctionType(
+            undefined,
+            [new IntType(256, false), new IntType(256, false), new IntType(256, false)],
+            [new IntType(256, false)],
             FunctionVisibility.Default,
             FunctionStateMutability.Pure
         )
     ],
     [
         "keccak256",
-        new SFunctionType(
-            [new SPointer(new SBytes(), DataLocation.Memory)],
-            [new SFixedBytes(32)],
+        new FunctionType(
+            undefined,
+            [new PointerType(new BytesType(), DataLocation.Memory)],
+            [new FixedBytesType(32)],
             FunctionVisibility.Default,
             FunctionStateMutability.Pure
         )
     ],
     [
         "sha256",
-        new SFunctionType(
-            [new SPointer(new SBytes(), DataLocation.Memory)],
-            [new SFixedBytes(32)],
+        new FunctionType(
+            undefined,
+            [new PointerType(new BytesType(), DataLocation.Memory)],
+            [new FixedBytesType(32)],
             FunctionVisibility.Default,
             FunctionStateMutability.Pure
         )
     ],
     [
         "ripemd160",
-        new SFunctionType(
-            [new SPointer(new SBytes(), DataLocation.Memory)],
-            [new SFixedBytes(20)],
+        new FunctionType(
+            undefined,
+            [new PointerType(new BytesType(), DataLocation.Memory)],
+            [new FixedBytesType(20)],
             FunctionVisibility.Default,
             FunctionStateMutability.Pure
         )
     ],
     [
         "ecrecover",
-        new SFunctionType(
-            [new SFixedBytes(32), new SIntType(8, false), new SFixedBytes(32), new SFixedBytes(32)],
-            [new SAddressType(false)],
+        new FunctionType(
+            undefined,
+            [
+                new FixedBytesType(32),
+                new IntType(8, false),
+                new FixedBytesType(32),
+                new FixedBytesType(32)
+            ],
+            [new AddressType(false)],
             FunctionVisibility.Default,
             FunctionStateMutability.Pure
         )
     ]
 ]);
 
-export const BuiltinAddressMembers = new Map<string, SType>([
-    ["balance", new SIntType(256, false)],
+export const BuiltinAddressMembers = new Map<string, TypeNode>([
```

Comment:
Missing `address.code` here I guess.

Reply:
I added it but there is an interesting problem here - scribble will now accept `address().code` on ALL solidity versions (including <0.8.0). And we will produce code that doesn't compile. So it seems that we need to modify this datastructure to make it aware of what versions ranges are supported for each.

Reply:
Fixed in f4281ca.